plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/compaction/CompactionOperation.java

```diff
@@ -0,0 +1,120 @@
+            List<String> deltaFileNames,
+            Optional<String> dataFileName,
+            HoodieFileGroupId id,
+            Map<String, Double> metrics,
+            Optional<String> bootstrapFilePath)
+    {
+        this.baseInstantTime = requireNonNull(baseInstantTime, "baseInstantTime is null");
+        this.dataFileCommitTime = requireNonNull(dataFileCommitTime, "dataFileCommitTime is null");
+        this.deltaFileNames = requireNonNull(deltaFileNames, "deltaFileNames is null");
+        this.dataFileName = requireNonNull(dataFileName, "dataFileName is null");
+        this.id = requireNonNull(id, "id is null");
+        this.metrics = requireNonNull(metrics, "metrics is null");
+        this.bootstrapFilePath = requireNonNull(bootstrapFilePath, "bootstrapFilePath is null");
+    }
+
+    public String getFileId()
+    {
+        return id.getFileId();
+    }
+
+    public String getPartitionPath()
+    {
+        return id.getPartitionPath();
+    }
+
+    public HoodieFileGroupId getFileGroupId()
+    {
+        return id;
+    }
+
+    public static CompactionOperation convertFromAvroRecordInstance(HoodieCompactionOperation operation)
+    {
+        Optional<String> dataFileName = Optional.ofNullable(operation.getDataFilePath());
+        return new CompactionOperation(
+                operation.getBaseInstantTime(),
+                dataFileName.map(path -> getCommitTime(new Path(path).getName())),
+                ImmutableList.copyOf(operation.getDeltaFilePaths()),
+                dataFileName,
+                new HoodieFileGroupId(operation.getPartitionPath(), operation.getFileId()),
+                operation.getMetrics() == null ? ImmutableMap.of() : ImmutableMap.copyOf(operation.getMetrics()),
+                Optional.ofNullable(operation.getBootstrapFilePath()));
+    }
+
+    @Override
+    public String toString()
+    {
+        return "CompactionOperation{baseInstantTime='" + baseInstantTime + '\'' + ", dataFileCommitTime="
+                + dataFileCommitTime + ", deltaFileNames=" + deltaFileNames + ", dataFileName=" + dataFileName + ", id='" + id
+                + '\'' + ", metrics=" + metrics + ", bootstrapFilePath=" + bootstrapFilePath + '}';
+    }
+
+    @Override
+    public boolean equals(Object o)
+    {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        CompactionOperation operation = (CompactionOperation) o;
+        return Objects.equals(baseInstantTime, operation.baseInstantTime)
+                && Objects.equals(dataFileCommitTime, operation.dataFileCommitTime)
+                && Objects.equals(deltaFileNames, operation.deltaFileNames)
+                && Objects.equals(dataFileName, operation.dataFileName) && Objects.equals(id, operation.id);
+    }
+
+    @Override
+    public int hashCode()
+    {
+        return Objects.hash(baseInstantTime, id);
```

Comment:
This seems broken. equals/hashcode use different fields. Are these even needed?

Reply:
I agree it seems broken but this is from their source code https://github.com/apache/hudi/blob/720c9d76c1d281b7a1305a205d195a4f179ce36a/hudi-common/src/main/java/org/apache/hudi/common/model/CompactionOperation.java#L156
I didn't want to change it.

Reply:
removed